### ISSUE TITLE
amp-carousel is not aggressive enough loading more slides

### ISSUE THREAD

**Issue by cramforce** [ampproject/amphtml#11797]
I believe that we currently only ever load the next slide. We should, while being conservative, load more pages. Especially on Desktop one can quickly click to a state where images are missing.

**Comment by ericlindley-g** [comment]
Sounds good— @camelburrito, is it easily configurable to change the number of slides before and after? If so, I might just try 2 (both before and after) and see how that feels. If it's not easy, I'd be curious to hear how involved this would be, so we can think about how to prioritize

**Comment by camelburrito** [comment]
@ericlindley-g The problem is not how many slides are hidden, it is howmany of them should we schedule to preload. And what is the  formula for selecting this

**Comment by ericlindley-g** [comment]
Right—updated my question to be more clear. Is that easy to experiment with, or something we should formally schedule?

**Comment by aghassemi** [comment]
@camelburrito @ericlindley-g I suggest using the same value (`renderOutsideViewport`) as runtime uses for scheduling. `renderOutsideViewport` defaults to `3` which can be interpreted as: "prerender if 3 or less slides away". Heavier components set it to lower, `amp-youtube` for instance sets it to `0.75` carousel and round that to `1`.

**Comment by cramforce** [comment]
+1 In an ideal world we'd schedule these resources as being N viewports away, but something more naive works for me.

**Comment by nainar** [comment]
This issue is fixed for all users who upgrade their `<amp-carousel>` version to 0.2. 

For more details please see documentation here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-carousel/amp-carousel.md
